docs/guides/BUILDING.md

```diff
@@ -31,12 +31,13 @@ To check out the Matter repository, run the following command:
 git clone --recurse-submodules git@github.com:project-chip/connectedhomeip.git
 ```
 
-## Synchronizing submodules
+## Updating Matter code
 
-If you already have the Matter code checked out, run the following command to
+If you already have the Matter code checked out, run the following commands to update the repository and 
 synchronize submodules:
 
 ```
+git pull
 git submodule update --init
```

Comment:
I wonder if `git submodule update -f --init --recursive` is better here. Sometimes I also needed a sync.

Reply:
True, I also have the "feeling" that it might be better, but I am not deep enough in git. Should I add it?

Reply:
I think you applied the change that you needed, if you never needed my call it is fine. If you ever need the recursive or -f please create a PR to add it here :)

Reply:
@andy31415 Fwiw, doing `--recursive` stuff on our submodules can get some of them in bad states, from what I have seen.  So it's actually sort of a bad idea.  :(

Reply:
Good to know :-) thank you!